modules/riemann/riemann.c

```diff
@@ -726,7 +718,6 @@ riemann_dd_new(GlobalConfig *cfg)
 
   self->super.worker.disconnect = riemann_dd_disconnect;
   self->super.worker.insert = riemann_worker_insert;
-  self->super.worker.thread_deinit = riemann_worker_thread_deinit;
```

Comment:
Maybe I don't understand, but does LogThreadedDestDriver really flush on deinit?
I don't see any registration of the `do_work` task which would eventually call `_perform_flush`.

Handling shutdown:
- I see `_shutdown_event_callback` event handler, which is called on deinit, but I only see task un-registering steps and no `do_work` scheduling or direct `flush` calling.
- I don't see any flush operation in `_worker_thread` either.

However, the original code with `riemann_worker_thread_deinit` calling `riemann_worker_batch_flush`  would not work either since `log_threaded_dest_worker_thread_deinit` is called after `_disconnect`.

As far as I see this concerns the http module as well.